### ISSUE TITLE
macro avg calculations

### ISSUE THREAD

**Issue by NoYo25** [chakki-works/seqeval#24]
Hi, I've used seqval in a NER system and it predicted me this classification report.  Screenshot  of the report is [here](https://drive.google.com/file/d/12SN6IJGr7cUwmeME8hfRWRXSaIM9lYg0/view?usp=sharing) 
Based on my knowledge, the macro avg of whatever metric (e.g. precision) would be in this case like the following
macro avg - pr = (0.74 + 0.85 + 0.96 + 0.65 + 0.87 + 0.54 + 0.33 + 0.00)  / 8 = 0.6175, which is not the case.

**Comment by kevinalexmathews** [comment]
It seems this issue has not been resolved yet. I would like to take a look at this bug.

**Comment by alanakbik** [comment]
It looks like the `average` parameter is currently ignored in all score methods: 
https://github.com/chakki-works/seqeval/blob/445d99798f6969f606dbf5978d936d5a8b33bbd1/seqeval/metrics/sequence_labeling.py#L116 

Are there plans to add macro average calculation to seqeval?

**Comment by davidggphy** [comment]
Same problem here. macro and micro give different answers, but macro value for sure is not correctly calculated

| precision    recall  f1-score   support            |
|----------------------------------------------------|
|                                                    |
| gpe       0.95      0.92      0.93      3159       |
| tim       0.83      0.82      0.83      4026       |
| geo       0.81      0.85      0.83      7527       |
| org       0.66      0.63      0.64      3980       |
| per       0.70      0.67      0.69      3395       |
| art       0.50      0.04      0.07        82       |
| eve       0.70      0.20      0.31        71       |
| nat       0.00      0.00      0.00        42       |
|                                                    |
| micro avg       0.79      0.78      0.79     22282 |
| macro avg       0.79      0.78      0.78     22282 |

**Comment by Querela** [comment]
It's only the micro average that is being computed if you look into the code.
I would guess the parameter is for future use or for compatibility with sklearn.
@developers But it might be better to throw an error if any value other than "micro" is supplied because the computation would be wrong or to just remove it as it is misleading ...
